entrypoint.sh

```diff
@@ -57,6 +58,14 @@ done
 echo $(date): ${http_status}
 MESSAGE="Spot Termination${CLUSTER_INFO}: ${NODE_NAME}, Instance: ${INSTANCE_ID}, AZ: ${AZ}"
 
+# If ASG_NAME is provided, detach the instance from ASG as soon as termination notice is available
+# This will stop new connections to this node, and make ASG launch the replacement node
+if [ "${ASG_NAME}" != "" ]; then
+  echo "Detaching INSTANCE ${INSTANCE_ID} FROM ASG ${ASG_NAME} and Region ${EC2_REGION}"
```

Comment:
Nitpick: "INSTANCE" and "FROM" should be lowercase (it's not SQL 😉 ) and "and" reads better as "in":
`echo "Detaching instance ${INSTANCE_ID} from ASG ${ASG_NAME} in Region ${EC2_REGION}"`